### ISSUE TITLE
Refactor variant_annotation

### ISSUE THREAD

**Issue by holtgrewe** [bihealth/snappy-pipeline#386]
**Is your feature request related to a problem? Please describe.**
The current `variant_annotation` step is broken.

**Describe the solution you'd like**
Refactor the step allowing for multiple tools and implement VEP only for now. We currently only need annotation with gnomAD frequency and ENSEMBL-based transcript annotation.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by ericblanc20** [comment]
@holtgrewe Two things:

- For cBioPortal export, we need additional control of the VEP parameters. In particular, we need to be able to pick the "best" transcript (`--pick` option) & other output options. For the latter, perhaps we could have a `output_options` parameter in the config file, with different defaults in `variant_annotation/__init__.py` & `somatic_variant_annotation/__init__.py`. For example:    
```
  output_options:
  # Germline pipeline
  - "terms SO"
  - symbol
  - hgvs
  - af_gnomade
  - af_gnomadg
  # Somatic pipeline
  - everything
```   
  The wrapper code would simply be (ugly, but just as an illustration):   
```
# In the python part:
current_step = snakemake.config["pipeline_step"]["name"]
vep_config = snakemake.config["step_config"][current_step]["vep"]
script_output_options = " ".join(["--" + x for x in vep["output_options"]])

...

# In the bash part:
vep --verbose --cache --offline --force_overwrite \
    --fork {vep_config[num_threads]} --buffer_size {vep_config[buffer_size]} \
    --cache_version {vep_config[cache_version]} --assembly {vep_config[assembly]} \
    $(if [[ ! -z "{vep_config[cache_dir]}" ]]; then \
        echo --dir_cache {vep_config[cache_dir]}; \
    fi) \
    {script_output_options} \
    $(if [[ "{vep_config[pick]}" = "yes" ]]; then \
        echo --pick; \
    fi) \
    --{vep_config[tx_flag]} \
    --fasta {snakemake.config[static_data_config][reference][path]} \
    --input_file {snakemake.input.vcf} --format vcf \
    --output_file {snakemake.output.vcf} --vcf --compress_output bgzip
```

As an additional benefit, it would make the wrapper more independent of the VEP version (for example, the options `af_gnomade` & `af_gnomadg` are not present in VEP version 102. gnomAD annotation is required by the option `af_gnomad`).

- We would like to use VEP & the associated cache for version 102. Because that version is used for GDC/TCGA, and I'd like to keep the annotation fairly consistent, and comparable to public studies if possible.

**Comment by ericblanc20** [comment]
Sure, I'll fix #384 now and worry about #387 later....
